src/shared/utilities/textUtilities.ts

```diff
@@ -235,3 +235,51 @@ export function getRelativeDate(from: Date, now: Date = new Date()): string {
     // if all conditionals above have failed, we're looking in terms of a > 1 year gap
     return rtf.format(Math.floor(fromAdj.getUTCFullYear() - now.getUTCFullYear()), 'year')
 }
+
+/**
+ * Format for rendering readable dates.
+ *
+ * Same format used in the S3 console, but it's also locale-aware.
+ * This specifically combines a separate date and time format
+ * in order to avoid a comma between the date and time.
+ *
+ * US: Jan 5, 2020 5:30:20 PM GMT-0700
+ * GB: 5 Jan 2020 17:30:20 GMT+0100
+ */
+export function getFormattedLocalizedDate(d: Date = new Date()): string {
+    const dateFormat = new Intl.DateTimeFormat('en-US', {
+        year: 'numeric',
+        month: 'short',
+        day: 'numeric',
+    })
+    const timeFormat = new Intl.DateTimeFormat('en-US', {
+        hour: 'numeric',
+        minute: 'numeric',
+        second: 'numeric',
+        timeZoneName: 'longOffset',
+    })
+
+    return `${dateFormat.format(d)} ${timeFormat.format(d)}`
+}
+/**
+ * Matches Insights console timestamp, e.g.: 2019-03-04T11:40:08.781-08:00
+ * TODO: Do we want this this verbose? Log stream just shows HH:mm:ss
```

Comment:
We could have separate functions, one for the date+time, the other for only time:

```
// HH:mm:ss
formatTimestamp

// 2019-03-04T11:40:08.781-08:00
formatDateTimestamp()
```

Reply:
This is a copy of the original comment. I have no opinion on the matter.